### PR TITLE
Develop, bubble-sort

### PR DESCRIPTION
**Pull request by kirokoNjenga**
**What does this PR do**
-The assignment was to create a bubble sort method that returns both the sorted array and block array
-We have created both files separately 
-We have test  everything is working as per the requirements
-Looking forward to your review

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice work :ok_hand: 
Your project looks good. However, we need to fix a few things before we can merge and proceed.
Check my comments for the suggestions.

### Martin Afani

File: bubble_sort_by.rb
Comment (whiz25) [inline bubble_sort_by.rb]: - [ ] bubble_sort_by method should be working **(covered by automated tests)** ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
- [ ] bubble_sort_by sorts an array by accepting a block
```ruby
words = %w[hi hey hello]
bubble_sort_by(words) { |left, right| left.length - right.length }
```
Should return `["hi", "hey", "hello"]` but instead it returns `[["hi", "hey", "hello"], 2]`

**Comment by kirokoNjenga** [comment]
**Recommended changes**
-worked on recommended change it working as required
image

File: bubble_sort_by.rb
Comment (stratospherique) [inline bubble_sort_by.rb]: - The same issue as mentioned by @whiz25 except that the method is currently returning `["hello", "hey", "hi"]`.

  - the `yield_` assignments lines in your method always take the second assignment as a final value, you should distinguish between the case where a block is given and the case where it's not.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Progress @team :clap: However, the bubble_sort_by method still needs further refactoring in order to get approval for the project. Check :arrow_down: 

- **[REQUIRED]** Check the code comments below. They should help you refactor the method :nerd_face: 

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by moinkhanif** [comment]
Tried to correct the required changes. If further update is required, please let us know. 

image

File: ruby-bubble/ruby-bubble-sort/bubblesort.rb
Comment (Forison) [inline ruby-bubble/ruby-bubble-sort/bubblesort.rb:1]: r = ["hi", "hello","hey"]
```print bubble_sort_by(r) {|left, right| right.length - left.length} = ["hi","hey","hello"]```
```print bubble_sort_by(r) {|left, right| right.length - left.length} != ["hello","hey","hi"]```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status: Approved 🛑
## 🙋‍♂️ Khan & Samuel
Good job so far, but the issue raised in the last review in relation to ```bubble_sort_by``` still persists. Check comment by @Ahmed and @whiz25
* [x] when given this ```["hi","hello","hey"]``` as arg ```bubble_sort_by``` should return ```["hi", "hey", "hello"]``` not ``` ["hello", "hey", "hi"] ```
[check the example in the pathright](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks)
* [x] remember, files related to the project should not be committed and added to .gitignore (e.g. .DS_Store) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
#### Happy coding
#### Addo Forison

File: ruby-bubble/ruby-bubble-sort/bubblesort.rb
Comment (moinkhanif) [reply ruby-bubble/ruby-bubble-sort/bubblesort.rb]: Hi Forinson,

Comment (moinkhanif) [reply ruby-bubble/ruby-bubble-sort/bubblesort.rb:1]: Hey there,

am sorry but in the project specifications, they have left.length - right.length and not right.length - left.length

Comment (moinkhanif) [reply ruby-bubble/ruby-bubble-sort/bubblesort.rb:1]: Could you please elaborate? because I just checked with the project specs and it works fine.
I used the following to test : 
```
print bubble_sort_by(%w[hi hello hey]) { |left, right|
  left.length - right.length
}
```

Comment (moinkhanif) [reply ruby-bubble/ruby-bubble-sort/bubblesort.rb:1]: even @whiz25 's code : 

```
words = %w[hi hey hello]

print bubble_sort_by(words) { |left, right| left.length - right.length }
```
works fine now. Could you please check again

**Comment by moinkhanif** [comment]
Hey there,

I just wanted to clarify that we did it according to the test example mentioned in the [Project specifications](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) :
Annotation 2020-02-19 023540

Therefore, 
```
words = %w[hey hi hello]

print bubble_sort_by(words) { |left, right| left.length - right.length }
```
will return:

`["hi","hey","hello"]`

**Review by stratospherique — APPROVED** [review]
### Great work applying the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**
